setup.py

```diff
@@ -29,7 +29,11 @@
         license="MIT",
         packages=find_packages(exclude=("test*", "examples")),
         include_package_data=True,
-        extras_require={"dev": development, "enum": ["marshmallow-enum~=1.5"]},
+        extras_require={
+            "dev": development,
+            "enum": ["marshmallow-enum~=1.5"],
+            "doc_parser": ["docstring_parser~=0.13"],
```

Comment:
It might be better to name this extra after the feature that it is providing. Though I'm not sure what exactly we should call this feature.
```suggestion
            "docstring-metadata": ["docstring_parser~=0.13"],
```